control/pid.go

```diff
@@ -24,23 +24,44 @@ func (l *Loop) newPID(config BlockConfig, logger logging.Logger) (Block, error)
 type basicPID struct {
 	mu       sync.Mutex
 	cfg      BlockConfig
-	error    float64
-	kI       float64
-	kD       float64
-	kP       float64
-	int      float64
+	PIDSets  []*PIDConfig
+	error    float64 // MIMO
+	kI       float64 //
+	kD       float64 //
+	kP       float64 //
+	int      float64 // MIMO
 	y        []*Signal
+	useMulti bool
 	satLimUp float64 `default:"255.0"`
 	limUp    float64 `default:"255.0"`
 	satLimLo float64
 	limLo    float64
 	tuner    pidTuner
-	tuning   bool
+	tuners   []*pidTuner
 	logger   logging.Logger
 }
 
+// GetTuning returns whether the PID block is currently tuning any signals
 func (p *basicPID) GetTuning() bool {
-	return p.tuning
+	// using locks so we do not check for tuning mid reconfigure or mid tune
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	return p.getTuning()
+}
```

Comment:
might be able to remove the locks from this. I added them because I was worried a reconfigure was breaking some logic, but i also don't hate keeping it?

Reply:
I think we can leave it if it's not causing issues. better to be safe

Reply:
okay thinking about this more, if we are locking here and then waiting one level up because of the locking, maybe we can safely get rid of both? if not it's definitely fine but something to maybe try? because the motor is always rebuild so it seems less important to lock in case of reconfigure issues but that might be a bad opinion. we can discuss it further but if you feel strongly towards keeping both, fine by me

Reply:
thinking about it more I want to keep the locks. since both the PID block and whoever is using `Get/MonitorTuning` are trying to access the same variables at the same, we need the lock to prevent race conditions.